python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -3053,6 +3057,7 @@ def test_all(data):
         [None, None],
         [[0, 5], [1, 6], [2, 7], [3, 8], [4, 9]],
         [[1, True], [2, False], [3, False]],
+        {},
```

Comment:
Split this condition off into a different test. That way we don't need to convolute this test with special casing logic based on inputs.

Reply:
Branched out into a separate test but kept the `axis` parameterization as we don't seem to be having tests for it.